processing/thermal.py

```diff
@@ -185,124 +172,122 @@ def update_metadata(conf, recording, api):
     api.update_metadata(recording, metadata, complete)
 
 
-def upload_tracks(
-    api, recording, main_model, model_results, logger, wallaby_device, master_name
-):
-    other_models = [model for model in model_results if model != main_model]
-    for track in main_model.tracks:
-        model_tags = []
-        track["id"] = api.add_track(recording, track, main_model.algorithm_id)
-        added = add_track_tags(api, recording, track, main_model, logger)
-        if added:
-            model_tags = [(main_model, track)]
-        # add track tags for all other models
-        for model in other_models:
-            track_to_save = find_matching_track(model.tracks, track)
-            if track_to_save is None:
-                logger.warn(
-                    "Could not find a matching track in model %s for recording %s track %s",
-                    model.model_config.name,
-                    recording["id"],
-                    track["id"],
-                )
-                continue
-            track_to_save["id"] = track["id"]
-            added = add_track_tags(api, recording, track_to_save, model, logger)
+def upload_tracks(api, recording, classify_result, wallaby_device, master_name, logger):
+    for track in classify_result.tracks:
+        track["id"] = api.add_track(
+            recording, track, classify_result.tracking_algorithm
+        )
+        model_results = []
+        for model_result in track["predictions"]:
+            added, tag = add_track_tag(api, recording, track, model_result, logger)
             if added:
-                model_tags.append((model, track_to_save))
+                model_results.append(
+                    (classify_result.models_by_id[model_result["id"]], model_result)
+                )
+        model_result = get_master_tag(model_results, wallaby_device)
```

Comment:
Reusing model_result variable name here is a bit confusing and could cause bugs.   Probably master_result or somethimg similar would be a better name.

Reply:
good point